commands/preupgrade/__init__.py

```diff
@@ -23,6 +23,11 @@
              help='Set preferred channel for the IPU target.',
              choices=['ga', 'tuv', 'e4s', 'eus', 'aus'],
              value_type=str.lower)  # This allows the choices to be case insensitive
+@command_opt('flavour', default=upgrade_path.LEAPP_UPGRADE_FLAVOUR_DEFAULT,
+             choices=[upgrade_path.LEAPP_UPGRADE_FLAVOUR_DEFAULT, upgrade_path.LEAPP_UPGRADE_FLAVOUR_SAP_HANA],
+             help='Specify upgrade path flavor')
+# NOTE(ivasilev) choices argument here will depend on flavor, so not specifying defaults
```

Comment:
@fernflower why the falvour option? I thought this is going to be detected automatically only.

Reply:
Yeah, that should be detected automatically. Don't see a case (apart some weird testing probably) when we would want to force SAP on a non sap machine. Will get rid of it, thanks.